src/content.ts

```diff
@@ -819,46 +821,58 @@ function getYouTubeVideoID(url: string) {
 /**
  * This function is required on mobile YouTube and will keep getting called whenever the preview bar disapears
  */
-function updatePreviewBarPositionMobile(parent: Element) {
+function updatePreviewBarPositionMobile(parent: HTMLElement) {
     if (document.getElementById("previewbar") === null) {
         previewBar.updatePosition(parent);
     }
 }
 
 function updatePreviewBar() {
-    if(isAdPlaying) {
-        previewBar.set([], [], 0);
+    if (previewBar === null) return;
+
+    if (isAdPlaying) {
+        previewBar.clear();
         return;
     }
 
-    if (previewBar === null || video === null) return;
+    if (video === null) return;
 
-    let localSponsorTimes = sponsorTimes;
-    if (localSponsorTimes == null) localSponsorTimes = [];
+    const previewBarSegments: PreviewBarSegment[] = [];
 
-    let allSponsorTimes = localSponsorTimes.concat(sponsorTimesSubmitting);
-	
-    //create an array of the sponsor types
-    let types = [];
-    for (let i = 0; i < localSponsorTimes.length; i++) {
-        if (localSponsorTimes[i].hidden === SponsorHideType.Visible) {
-            types.push(localSponsorTimes[i].category);
-        } else {
-            // Don't show this sponsor
-            types.push(null);
-        }
-    }
-    for (let i = 0; i < sponsorTimesSubmitting.length; i++) {
-        types.push("preview-" + sponsorTimesSubmitting[i].category);
+    if (sponsorTimes) {
+        sponsorTimes.forEach((segment) => {
+            if (segment.hidden !== SponsorHideType.Visible) return;
+
+            previewBarSegments.push({
+                timestamps: segment.segment as [number, number],
+                category: segment.category,
+                preview: false,
+            });
+        });
     }
 
-    previewBar.set(utils.getSegmentsFromSponsorTimes(allSponsorTimes), types, video.duration)
+    sponsorTimesSubmitting.forEach((segment) => {
+        previewBarSegments.push({
+            timestamps: segment.segment as [number, number],
```

Comment:
To be consistent, this should probably be `segment`

Reply:
I believe I named it `timestamps` instead of `segment` because the type itself is already called `Segment`. This results in slightly clunky naming such `segment.segment`. I also didn't want to name the `PreviewBarSegment` type `PreviewSponsorTime` even though it would be more consistent, because the name just seems awkward and confusing.

That said, this does make the naming within the project less consistent and perhaps this isn't the PR to fight it. Thoughts?

Reply:
I agree with you. I'll fix the others and commit to your branch

Reply:
So, it seems the old naming kind of has to stay since unsubmitted segments are stored using it and the API still uses it. The name "sponsorTime" at least should be able to change without issues though